src/integrator_interface.jl

```diff
@@ -4,26 +4,53 @@
 Update solution of `integrator`, if necessary or forced by `force_save`.
 """
 function savevalues!(integrator::DDEIntegrator, force_save=false)
-    # update ODE integrator
-    update_ode_integrator!(integrator)
+    # update time of ODE integrator (can be slightly modified (< 10ϵ) because of time stops)
+    if integrator.integrator.t != integrator.t
+        if abs(integrator.t - integrator.integrator.t) >= 10eps(integrator.t)
+            error("unexpected time discrepancy detected")
+        end
+
+        integrator.integrator.t = integrator.t
+        integrator.integrator.dt = integrator.integrator.t - integrator.integrator.tprev
+    end
 
-    # update solution of ODE integrator
+    # update solution
     savevalues!(integrator.integrator, force_save)
 
-    if typeof(integrator.sol.prob) <: ConstantLagDDEProblem
-        #warn("ConstantLagDDEProblem is deprecated. Use DDEProblem instead.")
-        neutral = false
-        constant_lags = integrator.sol.prob.lags
-    else
-        neutral = integrator.sol.prob.neutral
-        constant_lags = integrator.sol.prob.constant_lags
+    # copy additional interpolation steps back from solution to reduced ODE integrator
+    # last entry of solution always equals last time point of integration
+    if length(integrator.sol.k[end]) > length(integrator.integrator.k)
+        @inbounds for i in integrator.kshortsize+1:length(integrator.sol.k[end])
+            copyat_or_push!(integrator.integrator.k, i, integrator.sol.k[end][i])
+        end
+    end
+
+    # update prev2_idx to indices of tprev and u(tprev) in solution
+    integrator.prev2_idx = integrator.prev_idx
+
+    # cache dt of interval [tprev, t] of ODE integrator since it can only be retrieved by
+    # a possibly incorrect subtraction
+    integrator.integrator.dtcache = integrator.integrator.dt
```

Comment:
Will this make fixed timestep methods work incorrectly? `dtcache` already has a use.

Reply:
The `ODEIntegrator` is never iterated and besides inter/extrapolation only `savevalues!` and `postamble!` is applied to it. So I guess we can change everything that is not required for these tasks and methods without any effects - and hence we can also change `dtcache` arbitrarily.

Reply:
Comment it.